db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/HalfOuterJoinNode.java

```diff
@@ -923,6 +923,11 @@ public FromList flatten(ResultColumnList rcl,
             havingClause.remapColumnReferencesToExpressions();
         }
 
+        if (delayPreprocessingJoinClause()) {
```

Comment:
Does delayPreproessingJoinClause always return false for HalfOuterJoinNode?

Reply:
Yes, currently we don't have the logic to flatten ON clause subquery for outer joins (both Half and FullOuterJoinNode), so there is no need to delayPreprocessingJoinClause for Half and FullOuterJoinNode.

Reply:
If it is always false, this block will never executed.

Reply:
Hi @jyuanca You are correct, this block is not exercised now. I put it here in anticipation that in the future we may support the flattening of ON clause subquery for outer joins too.